engine/schema/src/com/cloud/host/dao/HostDaoImpl.java

```diff
@@ -995,8 +994,8 @@ public boolean updateState(Status oldStatus, Event event, Status newStatus, Host
             }
         }
         if (event.equals(Event.ManagementServerDown)) {
-            Float pingTimeout = NumbersUtil.parseFloat(_configDao.getValue("ping.timeout"), 2.5f);
-            Integer pingInterval = NumbersUtil.parseInt(_configDao.getValue("ping.interval"), 60);
+            float pingTimeout = (Float)_configDepot.get("ping.timeout").value();
```

Comment:
This still does not use ConfigKey?

Reply:
ConfigKey for ping.timeout and ping.interval is already defined in AgentManagerImpl. As per my understanding ConfigDepot cached all the configuration parameters so i am reading from configDepot.

Reply:
This switches from using one provider to another that uses that same strategy, if you use ConfigKey you won't have to specify the global setting name and handle the type-casting.

Reply:
@rhtyd 

ConfigKeys can only be defined once so as it is already defined and used in AgentManagerImpl, it can not be redefined. Also if you check the ConfigKeys, it read the configuration from ConfigDepot only.

Reply:
@mrunalinikankariya no you can refactor and move a `ConfigKey` to a new class that will cause change in its `category` that's all. Please test and try this approach instead.